src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -246,29 +251,84 @@ public Launcher decorate(final Launcher launcher, final Node node) {
             @Override
             public Proc launch(ProcStarter starter) throws IOException {
                 LOGGER.log(Level.FINEST, "Launch proc with environment: {0}", Arrays.toString(starter.envs()));
+
+                // find container working dir
+                KubernetesSlave slave = (KubernetesSlave) node;
+                FilePath containerWorkingDirFilePath = starter.pwd();
+                String containerWorkingDirStr = ContainerTemplate.DEFAULT_WORKING_DIR;
+                if (slave != null && slave.getPod().isPresent() && containerName != null) {
+                    Optional<Container> container = slave.getPod().get().getSpec().getContainers().stream()
+                            .filter(container1 -> container1.getName().equals(containerName))
+                            .findAny();
+                    Optional<String> containerWorkingDir = Optional.empty();
+                    if (container.isPresent() && container.get().getWorkingDir() != null) {
+                        containerWorkingDir = Optional.of(container.get().getWorkingDir());
+                    }
+                    if (containerWorkingDir.isPresent() && ! containerWorkingDirFilePath.getRemote().equals(containerWorkingDir.toString())) {
+                        // Container has a custom workingDir, set pwd to the container workspace root
+                        containerWorkingDirStr = containerWorkingDir.get();
+                        containerWorkingDirFilePath = new FilePath(containerWorkingDirFilePath.getChannel(), containerWorkingDirStr);
```

Comment:
Could be misleading, if the working directory has been changed using `dir('foo'){...}`, this will alter it to the container root working dir.

Reply:
I think I'm missing something. What do you mean by container root working dir? How is it different from `pwd`?

Reply:
if you have something like

```
container('foo') {
	dir('mydir') {
		sh 'pwd'
    }
}
```

Then mydir will be ignored with the currnet code.

Reply:
Got it, I'll update the groovy as suggested in the other comment, that will cover this use case and modify the code accordingly.

Reply:
What do we do if user has not set a `dir(`mydir`)`? In such case the `pwd` was pointing to `/home/jenkins/agent` which does not exist with in the container, should we change it in such a scenario?

Reply:
I think it should always be interpreted relatively to the current working dir.

So if you have `dir('foo')`, pwd should be `$WORKSPACE/foo`.